modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v4_3_5/DDMTemplateVersionUpgradeProcess.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.dynamic.data.mapping.internal.upgrade.v4_3_5;
+
+import com.liferay.portal.kernel.dao.jdbc.AutoBatchPreparedStatementUtil;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.Statement;
+
+/**
+ * @author Mariano Álvaro Sáiz
+ */
+public class DDMTemplateVersionUpgradeProcess extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		_upgradeDDMTemplateVersion();
+	}
+
+	private void _upgradeDDMTemplateVersion() throws Exception {
+		String selectOrphanTemplateVersions =
+			"select templateVersionId FROM DDMTemplateVersion where not " +
+				"exists (select * from DDMTemplate where " +
+					"DDMTemplate.templateId = DDMTemplateVersion.templateId)";
+
+		String deleteOrphanTemplateVersion =
+			"delete from DDMTemplateVersion where templateVersionId = ?";
+
+		try (Statement selectStatement = connection.createStatement();
```

Comment:
Hi @marianoalvarosaiz, why should we delete the other versions if the `DDMTemplate` still exists? Shouldn't we keep them for version control? Isn't that the purpose of the DDMTemplateVersion table?

Thanks!

Reply:
Hi @carolmariaabb,
 yes, that's why I checked **not exists** in _DDMTemplate_.
Kind regards.

Reply:
Got it now, thanks!